usermanager/src/main/java/org/pesmypetcare/usermanager/clients/pet/PetManagerClient.java

```diff
@@ -0,0 +1,480 @@
+    private static final String PETS_PICTURES_PATH = "/pets/";
+    private static final String PROFILE_IMAGE_NAME = "-profile-image.png";
+    private static final String VALUE_KEY = "value";
+    private static final String PUT = "PUT";
+    private static final String GET = "GET";
+    private static final String POST = "POST";
+    private static final String DELETE = "DELETE";
+    private static final String SLASH = "/";
+    private static final String NAME_FIELD = "name";
+    private final Gson GSON = new Gson();
+    private TaskManager taskManager;
+
+    public PetManagerClient() {
+        taskManager = new TaskManager();
+    }
+
+    /**
+     * Creates a pet entry in the data base for the user specified.
+     *
+     * @param accessToken The personal access token for the account
+     * @param username The user's username
+     * @param pet The pet's name
+     * @return The response code
+     * @throws ExecutionException When the retrieval of the pets fails
+     * @throws InterruptedException When the retrieval is interrupted
+     */
+    public int createPet(String accessToken, String username, Pet pet) throws ExecutionException, InterruptedException {
+        JSONObject reqJson = Pet.buildPetJsonObject(pet.getBody());
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(POST);
+        taskManager.setReqBody(reqJson);
+        StringBuilder response = taskManager
+                .execute(BASE_URL + PETS_PATH + username + SLASH + pet.getName(), accessToken).get();
+        return Integer.parseInt(response.toString());
+    }
+
+    /**
+     * Returns the data of a pet.
+     *
+     * @param accessToken The personal access token for the account
+     * @param username The pet's owner
+     * @param petName The pet's name
+     * @return The pet data
+     * @throws ExecutionException When the retrieval of the pet fails
+     * @throws InterruptedException When the retrieval is interrupted
+     */
+    public PetData getPet(String accessToken, String username, String petName)
+            throws ExecutionException, InterruptedException {
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder json = taskManager.execute(BASE_URL + PETS_PATH + username + SLASH + petName, accessToken).get();
+        if (json != null) {
+            return GSON.fromJson(json.toString(), PetData.class);
+        }
+        return null;
+    }
+
+    /**
+     * Returns all the pets of the user.
+     *
+     * @param accessToken The personal access token for the account
+     * @param username The user's username
+     * @return All the pets of the user
+     * @throws ExecutionException When the retrieval of the pets fails
+     * @throws InterruptedException When the retrieval is interrupted
+     */
+    public List<Pet> getAllPets(String accessToken, String username) throws ExecutionException, InterruptedException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [NCSS for this method is 17 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5561038)